src/test/java/com/synopsys/integration/jenkins/scan/factory/ScanParametersFactoryTest.java

```diff
@@ -0,0 +1,185 @@
+package com.synopsys.integration.jenkins.scan.factory;
+
+import com.synopsys.integration.jenkins.scan.exception.PluginExceptionHandler;
+import com.synopsys.integration.jenkins.scan.extension.pipeline.SecurityScanStep;
+import com.synopsys.integration.jenkins.scan.global.ApplicationConstants;
+import hudson.AbortException;
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.model.TaskListener;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import java.io.File;
+import java.io.PrintStream;
+import java.util.HashMap;
+import java.util.Map;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ScanParametersFactoryTest {
+    private TaskListener listenerMock;
+    private FilePath workspace;
+    private EnvVars envVarsMock;
+    private SecurityScanStep securityScanStep;
+
+    @BeforeEach
+    public void setUp() {
+        workspace = new FilePath(new File(System.getProperty("user.home")));
+        listenerMock = Mockito.mock(TaskListener.class);
+        envVarsMock = Mockito.mock(EnvVars.class);
+        securityScanStep = new SecurityScanStep();
+        Mockito.when(listenerMock.getLogger()).thenReturn(Mockito.mock(PrintStream.class));
+    }
+
+    @Test
+    public void preparePipelineParametersMapTest() throws AbortException, PluginExceptionHandler {
+        Map<String, Object> globalConfigValues = new HashMap<>();
+        ScanParametersFactory scanParametersFactory = new ScanParametersFactory(envVarsMock, workspace);
+
+        securityScanStep.setProduct("BLACKDUCK");
+        securityScanStep.setBitbucket_token("FAKETOKEN");
+        globalConfigValues.put(ApplicationConstants.BLACKDUCK_URL_KEY, "https://fake-blackduck.url");
+        globalConfigValues.put(ApplicationConstants.BLACKDUCK_TOKEN_KEY, "fake-blackduck-token");
+        globalConfigValues.put(ApplicationConstants.SYNOPSYS_BRIDGE_INSTALL_DIRECTORY, "/fake/path");
+
+        Map<String, Object> result = scanParametersFactory.preparePipelineParametersMap(securityScanStep,
+                globalConfigValues, workspace, listenerMock);
+
+        assertEquals(5, result.size());
+        assertEquals("BLACKDUCK", result.get(ApplicationConstants.PRODUCT_KEY));
+        assertEquals("fake-blackduck-token", result.get(ApplicationConstants.BLACKDUCK_TOKEN_KEY));
+        assertEquals("/fake/path", result.get(ApplicationConstants.SYNOPSYS_BRIDGE_INSTALL_DIRECTORY));
+        assertEquals("FAKETOKEN", result.get(ApplicationConstants.BITBUCKET_TOKEN_KEY));
+
+        securityScanStep.setProduct("invalid-product");
+
+        assertThrows(PluginExceptionHandler.class, () -> scanParametersFactory.preparePipelineParametersMap(securityScanStep,
```

Comment:
Here, preparePipelineParametersMap static method is accessed by the instance reference. I think static method should be accessed by the class name.